### ISSUE TITLE
Buttons have no ID optionality

### ISSUE THREAD

**Issue by Sven-Q** [PaulNieuwelaar/alertjs#27]
Hello,

I am using Version 2.1 of These Dialogs in my CRM System. However the Dialogs work great I feel it is missing some core functionality.

In some use cases I have to disable the Buttons untill some requirements have been met, for this i Need to have an ID on the Buttons, but there is no way for me to achieve this at the Moment or am I missing a feature?

I would be able to do it by requesting all the button Elements inside #alertJS-tdDialogFooter, however this is not very secure due to having the possibilty of this id changing in future Releases/bugfixes.

With Kind regards,
SvenQ

**Comment by PaulNieuwelaar** [comment]
Hi Sven,

Unfortunately we currently don't support disabling the alert buttons. As you mentioned it's technically possible with something like this: `Alert.$("button:contains('OK')").prop("disabled", true)` but there's no guarantee that the buttons will still be buttons in the future etc, so do this at your own risk.

The way we usually recommend adding validation is to keep the buttons clickable, but add the validation on click. If some input on the popup is invalid then you can display the error message to the user and prevent the button action. Note that you can use the [preventClose](https://github.com/PaulNieuwelaar/alertjs/wiki/Documentation#alertshow) property on the button object to keep the alert open so you can run validation, then manually close it with `Alert.hide();` when the validation is successful.

You may also want to consider upgrading to v3.0 as the support for Alert.$ and nested alerts are only in 3.0.

Cheers,
Paul

**Comment by Sven-Q** [comment]
Hi Paul,

Thanks for the quick Response.

At the Moment we have fixed it by keeping the custom Buttons (with Code) in the HTML itself and not defining them in the constructor. Afterwards we then close the Dialog ourselves.

I only made this Issue, due to that it actually feels like this is a missing feature in the great product you provide and which could Maybe eaily be implemented in the button constructor?

Anyway, I thank you once more for the quick Response and the great product you provide.

Regards,
Sven
